Code/GraphMol/ChemReactions/ReactionRunner.cpp

```diff
@@ -854,4 +882,192 @@ std::vector<MOL_SPTR_VECT> run_Reactants(const ChemicalReaction &rxn,
   return productMols;
 }  // end of ChemicalReaction::runReactants()
 
+// Generate a product based on a SINGLE reactant
+std::vector<MOL_SPTR_VECT> run_Reactant(const ChemicalReaction &rxn,
+                                        const ROMOL_SPTR &reactant,
+                                        unsigned int reactantIdx) {
```

Comment:
Can we have a heuristic that figures out the reactantIdx automatically?  Maybe some heuristic such as using the MCS of the reactant with each of the reactant's rxns